src/main/java/jp/co/future/uroborosql/tx/LocalTransactionManager.java

```diff
@@ -391,4 +393,33 @@ public void rollback() {
 		}
 	}
 
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.tx.TransactionManager#savepointScope(jp.co.future.uroborosql.tx.SQLSupplier)
+	 */
+	@Override
+	public <R> R savepointScope(final SQLSupplier<R> supplier) {
+		String savepointName = UUID.randomUUID().toString();
+		setSavepoint(savepointName);
+		try {
+			return supplier.get();
+		} catch (UroborosqlRuntimeException ex) {
```

Comment:
@HidekiSugimoto189 
Here you should catch on `Exception`, not `UroborosqlRuntimeException.`
The user does not know what exception to throw.

Reply:
Modified to catch throwable.
The runInNewTx() method has also been changed to catch throwable.

Reply:
@shout-star 
Please review.